starboard/android/apk/app/src/main/java/dev/cobalt/media/VideoSurfaceView.java

```diff
@@ -75,6 +78,19 @@ private void initialize(Context context) {
     // punch-out video when the position / size is animated.
   }
 
+  public void clearSurface() {
+    if (getHolder().getSurface().isValid()) {
+      Canvas canvas = getHolder().lockCanvas();
+      if (canvas != null) {
```

Comment:
Would we want to return when canvas is null? Do we still need to run setFormat() twice when it is null?

Reply:
I added the check as a precaution to avoid crashing from dereferencing a null if `lockCanvas()` can ever fail, and if that happens the clearing of the surface obviously isn't happening, but the dangling-lock side-effect that results in the `already connected` issue could in theory actually still happen, in which case we do want to do this transparent + opaque swap to entice the lock to be released, otherwise the user will see a playback error. There also isn't a guarantee that those statements would fail, so I think we can assume they will work as they normally do.

Reply:
https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/core/java/com/android/internal/view/BaseSurfaceHolder.java;l=222?q=lockCanvas
The code here indicates when null is returned, it is not locked.

Reply:
Wait a second, the error is not about the lock.